### PR TITLE
Vcpkg manifest build + ASan and UBSan

### PR DESCRIPTION
**Pull request by byronxu99**
- Build rlclientlib and binary parser using vcpkg and CMake presets
- Build with AddressSanitizer and UndefinedBehaviorSanitizer
- Add CI workflows to run tests with sanitizers
- Fix bugs found by sanitizers

### CODE REVIEW

**Comment by olgavrou** [comment]
we should start updating vw submodule based on tagged releases

we should have another release coming up early September so we should plan to update VW then

**Comment by byronxu99** [comment]
> we should start updating vw submodule based on tagged releases

Yeah that's a good idea. This PR can wait until then to get merged.

**Comment by byronxu99** [comment]
"Integration with latest VW" check may fail for now due to formatting changes on the master branch of VW.

However, there shouldn't be any issues in building rlclientlib with VW release 9.4.0.

File: CMakePresets.json
Comment (olgavrou) [inline CMakePresets.json:16]: we have submoduled vcpkg in rlclientlib, can we point to that `vcpkg.cmake` instead of vw's?

**Review by olgavrou — APPROVED** [review]
approved to be merged after the 9.4 update

**Pull request merged**
